xgcm/grid.py

```diff
@@ -1268,19 +1268,46 @@ def set_metrics(self, key, value):
                 f"Metric axes {axes_not_found!r} not compatible with grid axes {tuple(self.axes)!r}"
             )
 
-        # initialize empty list
-        self._metrics[metric_axes] = []
-        for metric_varname in _maybe_promote_str_to_list(value):
-            if metric_varname not in self._ds:
-                raise KeyError(
-                    f"Metric variable {metric_varname} not found in dataset."
-                )
-            # resetting coords avoids potential broadcasting / alignment issues
-            metric_var = self._ds[metric_varname].reset_coords(drop=True)
+        metric_value = _maybe_promote_str_to_list(value)
+        existing_metric_axes = set(k for k in self._metrics.keys())
 
-            # TODO: check for consistency of metric_var dims with axis dims
-            # check for duplicate dimensions among each axis metric
-            self._metrics[metric_axes].append(metric_var)
+        if metric_axes in existing_metric_axes:
+            value_exist = self._metrics.get(metric_axes)
+            for metric_varname in metric_value:
+                if metric_varname not in self._ds:
+                    raise KeyError(
+                        f"Metric variable {metric_varname} not found in dataset."
+                    )
+                else:
+                    value_new = self._ds[metric_varname].reset_coords(drop=True)
+                    did_overwrite = False
+                    # go through each existing value until data array with matching dimensions is selected
+                    for idx, ve in enumerate(value_exist):
+                        # double check if dimensions match
+                        if set(value_new.dims) == set(ve.dims):
+                            if overwrite:
+                                # replace existing data array with new data array input
+                                self._metrics[metric_axes][idx] = value_new
+                                did_overwrite = True
+                            else:
+                                raise ValueError(
+                                    f"Metric variable {metric_varname} with dimensions {ve.dims} already assigned in metrics."
+                                    f" Overwrite {metric_varname} by setting overwrite=True."
+                                )
+                    # if no existing value matches new value dimension-wise, just append new value
+                    if not did_overwrite:
+                        self._metrics[metric_axes].append(value_new)
+        else:
+            # initialize empty list
```

Comment:
```suggestion
            # no metrics for metric_axes yet; initialize empty list
```

I was confused about when this else is being triggered. Another reason to reduce indentation above,